sstables/sstables.cc

```diff
@@ -507,7 +507,7 @@ future<> parse(const schema& schema, sstable_version_types v, random_access_read
 
     // Positions are encoded in little-endian.
     auto b = buf.get();
-    s.positions = utils::chunked_vector<pos_type>();
+    s.positions.reserve(s.header.size + 1);
```

Comment:
This looks like we wanted to reset the positions to empty vector, in case it had content.
I see we don't do this consistently with all members, but maybe we should still keep the reset just in case.

Reply:
okay, I'll make sure it's still needed.

Reply:
I don't see where the summary component is reused, so I don't think this resetting is required.
We only call parse from https://github.com/scylladb/scylladb/blob/cd2a2bd02143598cc6e88ad2b99f6ed7301355ba/sstables/sstables.cc#L1318
After checking https://github.com/scylladb/scylladb/blob/cd2a2bd02143598cc6e88ad2b99f6ed7301355ba/sstables/sstables.cc#L1308-L1310

So we don't reuse the object.

Reply:
I'd suggest to add an explanation in the commit why it can go away. I'd expect a separate commit for removing the reset, but maybe that's bikeshedding since it's a simple change.

Reply:
I would add a comment in front of this function saying it can only be called once.
But, wouldn't it be easier just to allow it to be called twice? I.e., leaves the reset as in the old code and then add the reserve() you wanted to add?

Reply:
But we didn't reset `s.entries` before so it was never supposed to work this way.
We don't need to invent new use cases to excuse some old artifact.

Reply:
That's a good point.